lib/tasks/data_feeds/dark_sky_temperature_loader.rake

```diff
@@ -0,0 +1,37 @@
+namespace :data_feeds do
+  def process_area(area, start_date, end_date)
+    data = DarkSkyWeatherInterface.new.historic_temperatures(
+      area.latitude,
+      area.longitude,
+      start_date,
+      end_date
+    )
+
+    # Data is returned as an array [[distance_to_weather_station, temperature_data, percent_bad, bad_data]
+    temperature_data = data[1]
+
+    temperature_data.each do |reading_date, temperature_celsius_x48|
+      next if temperature_celsius_x48.size != 48
+      record = DataFeeds::DarkSkyTemperatureReading.find_by(reading_date: reading_date, area_id: area.id)
+      if record
+        record.update(temperature_celsius_x48: temperature_celsius_x48)
```

Comment:
Could we use an upsert here?

Reply:
We could - I wanted to look at the Rails 6 upsert functionality at some point.